deployment/common/changeset/deploy_link_token.go

```diff
@@ -110,10 +114,25 @@ func deployLinkTokenContractEVM(
 ) (*deployment.ContractDeploy[*link_token.LinkToken], error) {
 	linkToken, err := deployment.DeployContract[*link_token.LinkToken](lggr, chain, ab,
 		func(chain deployment.Chain) deployment.ContractDeploy[*link_token.LinkToken] {
-			linkTokenAddr, tx, linkToken, err2 := link_token.DeployLinkToken(
-				chain.DeployerKey,
-				chain.Client,
+			var (
+				linkTokenAddr common.Address
+				tx            *eth_types.Transaction
+				linkToken     *link_token.LinkToken
+				err2          error
 			)
+			if !chain.IsZk {
+				linkTokenAddr, tx, linkToken, err2 = link_token.DeployLinkToken(
+					chain.DeployerKey,
+					chain.Client,
+				)
+			} else {
+				linkTokenAddr, _, linkToken, err2 = link_token.DeployLinkTokenZk(
```

Comment:
if tx is remains nil here, is it not creating any np issue further?

Reply:
We are not using the result any further. See https://github.com/smartcontractkit/chainlink/blob/02277eca03033e11a8c710d3cb05ec89d60d6e0c/deployment/common/changeset/deploy_link_token.go#L50

The tx is used only in the `Confirm`